### ISSUE TITLE
Privacy plugin breaks on files stored in memory

### ISSUE THREAD

**Issue by landmaj** [squidfunk/mkdocs-material#7803]
### Context

The bug was encountered by a user of mkdocs-d2-plugin. Relevant issue: https://github.com/landmaj/mkdocs-d2-plugin/issues/21

### Bug description

The privacy plugin [passes `abs_src_path` to `open()` without checking for `None`](https://github.com/squidfunk/mkdocs-material/blob/7fb3a39bbc17865743b8cb8846aa4e845ac53eb4/src/plugins/privacy/plugin.py#L273), which results in an error:

```
TypeError: expected str, bytes or os.PathLike object, not NoneType
```

`None` is a valid value since MkDocs 1.6, as described [here](https://www.mkdocs.org/dev-guide/api/#mkdocs.structure.files.File):

> Since MkDocs 1.6 a file may alternatively be stored in memory - `content_string`/`content_bytes`.
>
> Then `src_dir` and `abs_src_path` will remain `None`. `content_bytes`/`content_string` need to be written to, or populated through the `content` argument in the constructor.

The (valid) code in my plugin which causes privacy plugin to crash is this:
```python
  def on_files(self, files: Files, config):
      content = importlib_files("d2.css").joinpath("mkdocs_d2_plugin.css").read_text()
      file = File(
          "assets/stylesheets/mkdocs_d2_plugin.css",
          None,
          config["site_dir"],
          config["use_directory_urls"],
      )
      file.content_string = content

      files.append(file)
```

### Related links

- [Reporting a bug](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/)

### Reproduction

[9.5.48-privacy-plugin-crash.zip](https://github.com/user-attachments/files/18147854/9.5.48-privacy-plugin-crash.zip)

### Steps to reproduce

1. Run `mkdocs build` with `privacy` and `d2` plugins enabled.

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material/), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/) using the [built-in info plugin](https://squidfunk.github.io/mkdocs-material/plugins/info/).

**Comment by squidfunk** [comment]
Thanks for reporting. Fixed in 968fbe1ae213cb350850cfddf2614a1c93e73adb. To be honest, a fundamental change like this to MkDocs' plugin API would actually mandate a breaking change, IMHO. We're probably not be the only plugin that is impacted.

**Comment by squidfunk** [comment]
Released as part of [9.5.49](https://squidfunk.github.io/mkdocs-material/changelog/#9.5.49).
